apis/refs/v1beta1/kmskeyringref.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// +reference-type
+// apiVersion: kms.cnrm.cloud.google.com/v1beta1
+// kind: KMSKeyRing
+// external: projects/{project}/locations/{location}/keyRings/{name}
+
+type KMSKeyRingRef struct {
+	/* The keyRing selflink of form "projects/{project}/locations/{location}/keyRings/{name}", when not managed by KCC. */
+	External string `json:"external,omitempty"`
+	/* The `name` field of a `KMSKeyRing` resource. */
+	Name string `json:"name,omitempty"`
+	/* The `namespace` field of a `KMSKeyRing` resource. */
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
(I shall ask this earlier) What's the (current) plan referencing to a different namespace?